framework/include/utils/DelimitedFileReader.h

```diff
@@ -32,7 +32,8 @@ namespace MooseUtils
  * Communicator is provide then it will only read on processor 0 and broadcast the data to all
  * processors. If not provided it will read on all processors.
  */
-class DelimitedFileReader
+template <typename T>
+class DelimitedFileReaderTempl
```

Comment:
Irrelevant to this PR: I guess this is a good first step towards mixed type csv reader, i.e., different data type for each column. Isn't there an open-source c++ csv reader?